web/src/engine/websites/decorators/CoreView.ts

```diff
@@ -0,0 +1,261 @@
+    const uri = new URL(data[0].href, this.URI);
+    uri.searchParams.set('free_only', '0'); // 0: include non-free, 1: exclude non-free
+    data = await FetchCSS(new FetchRequest(uri.href), query);
+    return data.map(element => {
+        const { id, title } = extractor.call(this, element, manga);
+        return new Chapter(this, manga, id, title);
+    });
+}
+
+/**
+ * A class decorator that adds the ability to extract all chapters for a given manga from this website using the given CSS {@link query}.
+ * The chapters are extracted from the composed url based on the `Identifier` of the manga and the `URI` of the website.
+ * @param query - A CSS query to locate the elements from which the chapter identifier and title shall be extracted
+ * @param extractor - An Extractor to get chapter infos
+ */
+export function ChaptersSinglePageCSS(query: string = queryChapters, extractor = ChapterExtractor) {
+    return function DecorateClass<T extends Common.Constructor>(ctor: T, context?: ClassDecoratorContext): T {
+        if (context && context.kind !== 'class') {
+            throw new Error(context.name);
+        }
+        return class extends ctor {
+            public async FetchChapters(this: MangaScraper, manga: Manga): Promise<Chapter[]> {
+                return FetchChaptersSinglePageCSS.call(this, manga, query, extractor);
+            }
+        };
+    };
+}
+
+/**********************************************
+ ******** Page List Extraction Methods ********
+ **********************************************/
+
+/**
+ * An extension method for extracting all pages for the given {@link chapter} using the given CSS {@link query}.
+ * The pages are extracted from the composed url based on the `Identifier` of the {@link chapter} and the `URI` of the website.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param chapter - A reference to the {@link Chapter} which shall be assigned as parent for the extracted pages
+ * @param query - A CSS query to locate the JSON from which the page information shall be extracted
+ */
+async function FetchPagesSinglePageJSON(this: MangaScraper, chapter: Chapter, query = queryEpisodeJSON): Promise<Page[]> {
+    const request = new FetchRequest(new URL(chapter.Identifier, this.URI).href);
+    const dataElement = await FetchCSS(request, query);
+    const data: ChapterJSON = JSON.parse(dataElement[0].dataset.value);
+    if (!data.readableProduct.isPublic && !data.readableProduct.hasPurchased) {
+        throw new Error(`The chapter '${chapter.Title}' is neither public, nor purchased!`);
+    }
+    return data.readableProduct.pageStructure.pages.filter(page => page.type === 'main').map(page => {
+        // NOTE: 'usagi' is not scrambled
+        let parameters = {};
+        if (data.readableProduct.pageStructure.choJuGiga === 'baku') {
+            parameters = { Referer: this.URI.href, scrambled: 1 };
```

Comment:
The parameter scrambled is numeric, but seems used like a boolean.
Consider to use boolean instead
`parameters = { Referer: this.URI.href, scrambled: data.readableProduct.pageStructure.choJuGiga === 'baku' }`

Reply:
Boolean is not supported in parameters, otherwise i would have use a boolean ;).
Should i modify the file to allow booleans?